pandas/core/groupby/generic.py

```diff
@@ -1863,6 +1909,30 @@ def groupby_series(obj, col=None):
             results.index = ibase.default_index(len(results))
         return results
 
+    @doc(_agg_template, fname="sum", no=True, mc=0, return_type="DataFrame")
+    def sum(self, numeric_only: bool = True, min_count: int = 0) -> DataFrame:
+        return super().sum(numeric_only=numeric_only, min_count=min_count)
+
+    @doc(_agg_template, fname="prod", no=True, mc=0, return_type="DataFrame")
+    def prod(self, numeric_only: bool = True, min_count: int = 0) -> DataFrame:
+        return super().prod(numeric_only=numeric_only, min_count=min_count)
+
+    @doc(_agg_template, fname="min", no=False, mc=-1, return_type="DataFrame")
+    def min(self, numeric_only: bool = False, min_count: int = -1) -> DataFrame:
+        return super().min(numeric_only=numeric_only, min_count=min_count)
+
+    @doc(_agg_template, fname="max", no=False, mc=-1, return_type="DataFrame")
+    def max(self, numeric_only: bool = False, min_count: int = -1) -> DataFrame:
+        return super().max(numeric_only=numeric_only, min_count=min_count)
+
+    @doc(_agg_template, fname="first", no=False, mc=-1, return_type="DataFrame")
+    def first(self, numeric_only: bool = False, min_count: int = -1) -> DataFrame:
+        return super().first(numeric_only=numeric_only, min_count=min_count)
+
+    @doc(_agg_template, fname="last", no=False, mc=-1, return_type="DataFrame")
+    def last(self, numeric_only: bool = False, min_count: int = -1) -> DataFrame:
+        return super().last(numeric_only=numeric_only, min_count=min_count)
+
```

Comment:
These are duplicates from above except the return values. Can you use Generic instead? The exception would be for unique docstrings for the Series/Frame methods. Does something need to be added to make these docstrings visible in the documentation.

Reply:
The doc string parameter ``return_type`` is also different, though. I could change these into

```python
@doc(_agg_template, fname="last", no=False, mc=-1, return_type="DataFrame")
def last(self, numeric_only: bool = False, min_count: int = -1) -> DataFrame:
        last_compat = partial(self._get_loc, loc=-1)

        return self._agg_general(
            numeric_only=numeric_only,
            min_count=min_count,
            alias="last",
            npfunc=last_compat,
        )
```

on both SeriesGroupby and DataFrameGroupby (setting 'Dataframe' and 'Series' the relevant places). That would be repetitipus also, though, but it would avoid setting methods on the NDFrame class...

Reply:
sorry if I wasn't clear, in the docs I see https://pandas.io/docs/reference/api/pandas.core.groupby.GroupBy.min.html but nothing for DataFrameGroupBy.min, whereas for most other methods there is a DataFrameGroupBy.<method>

So my question is, how to make these visible, otherwise if the generic docstring is used, this duplication is unnecessary and GroupBy could be Generic to give the correct return type. see https://github.com/pandas-dev/pandas/pull/29480/files#r347723361 where the idea was rejected.

Reply:
Ok, I see your point now, it's the issue of the doc reference.

I don't have a good solution to that. I've added doc strings to GroupBy methods and copied them over to SeriesGroupby and DataFrameGroupby. SeriesGroupby and DataFrameGroupby need methods themselves in order to get the correct type hint. feels repetitive, but I don't think we can make it better without dropping the type hint.